Assets/Scripts/RoadControl.cs

```diff
@@ -355,125 +591,467 @@ void Update()
+                    curWarningRoadLineIndex++;
                 }
-            }
 
-            if (visibleDroids.Count > 0) {
-                foreach (BombDroid bd in visibleDroids) {
+                if (segNumDrawn < segIndexSkipSegThreshold) {
+                    if (visibleBillboards.Count > 0) {
+                        foreach (BillboardSprite bb in visibleBillboards) {
+                            if (bb.segmentIndex == i) {
+
+                                float roadWidthOffset;
+                                if (bb.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
+                                else { roadWidthOffset = screenHalfWidth; }
 
-                    if (bd.segmentIndex == i) {
+                                float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
 
-                        float roadWidthOffset;
-                        if (bd.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
-                        else { roadWidthOffset = screenHalfWidth; }
+                                float spriteZ;
+                                float spriteElevation = nearEdgeHeight + bb.spriteType.rect.height * nearEdgeWidthScale / 2;
+                                if (nearEdgeHeight < absHighestScreenLineDrawn && roadSegIndexOfHighestScreenLine < i) { spriteZ = spriteBgdZ; }
+                                else { spriteZ = -1f; }
 
-                        float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
+                                bb.spriteTransform.localScale = new Vector3(sizeCorrection, sizeCorrection, 1);
+                                bb.spriteTransform.position = new Vector3(x + (bb.offsetX) * nearEdgeWidthScale,
+                                                                            spriteElevation, spriteZ);
+                                bb.spriteRend.enabled = true;
+                            }
+                        }
+                    }
 
-                        bd.droid.localScale = new Vector3(sizeCorrection, sizeCorrection, 1);
-                        bd.droid.position = new Vector3(x + (bd.offsetX) * nearEdgeWidthScale,
-                                                                    nearEdgeHeight + (droidHeight + bd.droidSprite.sprite.rect.height) * nearEdgeWidthScale / 2, -1f);
-                        bd.droidSprite.enabled = true;
+                    if (visibleDroids.Count > 0) {
+                        foreach (BombDroid bd in visibleDroids) {
+
+                            if (bd.segmentIndex == i) {
+
+                                float roadWidthOffset;
+                                if (bd.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
+                                else { roadWidthOffset = screenHalfWidth; }
+
+                                float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
```

Comment:
`float roadWidthOffset;`
`float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;`

-can be moved up to the outer most `if` so that they can be reused by the lower sibling conditionals instead of being re-declared in different places.